### PR TITLE
[feat]: [CDS-73149]: adding azure sdk network interceptor for logging request/response URLs

### PR DESCRIPTION
**Pull request by markoharness**
## Describe your changes
Removed all trace logs for azure swimline that were logging out sensitive info
Added azure sdk network interceptor to log request/response URLs

Started up a delegate and tested on QA env

Screenshot 2023-06-28 at 15 15 05

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49821)

### CODE REVIEW

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger publish-delegate

**Comment by markoharness** [comment]
trigger uts

File: 960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class LoggingInterceptor implements Interceptor {
+  @NotNull
+  @Override
+  public Response intercept(@NotNull Chain chain) throws IOException {
+    Request request = chain.request();
+
+    long t1 = System.nanoTime();
+
+    if (request != null && request.url() != null && chain != null && chain.connection() != null
+        && request.headers() != null) {
```
Comment (yogesh-chauhan) [inline 960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java:31]: will headers always need to be present for this logging to be helpful ?

**Comment by markoharness** [comment]
*[`960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/49821#gh-1244172438:-NYyARLG84C54en8B6R8:b5l2hxp) ([raw file](https://github.com/harness/harness-core/blob/1ca6120bf9a11c8be765c14621e5eca17f4bfe6a/960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java#L31)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

will headers always need to be present for this logging to be helpful ?
</blockquote></details>

No. i will change the logging to be in debug mode (so that it can be use only when necessary). So that way we can leave the info we are logging as it is. What do you think?

**Comment by markoharness** [comment]
trigger publish-delegate

**Comment by markoharness** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]

**Review by vlica-harness — DISMISSED** [review]

**Review by ivan-83 — CHANGES_REQUESTED** [review]
Looks good with minor comments related to the async client instance.

> Reviewed 5 of 7 files at r1, 1 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49821#-:-NZ39rwu2uooMMQ3zJsQ:b-z0tvci)* status: 7 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java` line 347 at r4](https://reviewable.io/reviews/harness/harness-core/49821#-NZ37RUQ6Zuu8pYylVVg:-NZ37RUQ6Zuu8pYylVVh:bce19dt) ([raw file](https://github.com/harness/harness-core/blob/64f419a4bff7e0362b778b9bfd6828fcb953a2cb/960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java#L347)):*
> ```Java
>                   .readTimeout(clientOptions.getReadTimeout());
>     Integer poolSize = clientOptions.getMaximumConnectionPoolSize();
>     int maximumConnectionPoolSize = poolSize != null && poolSize > 0 ? poolSize : 5;
> ```

Do we need a connection pool for each instance of the async client? Should we manage this connection pool, close and terminate connection pool threads when the API call is done?
___
*[`960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java` line 347 at r4](https://reviewable.io/reviews/harness/harness-core/49821#-NZ37caz5UhNujQk_ka1:-NZ37caz5UhNujQk_ka2:bmeof7t) ([raw file](https://github.com/harness/harness-core/blob/64f419a4bff7e0362b778b9bfd6828fcb953a2cb/960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java#L347)):*
> ```Java
>                   .readTimeout(clientOptions.getReadTimeout());
>     Integer poolSize = clientOptions.getMaximumConnectionPoolSize();
>     int maximumConnectionPoolSize = poolSize != null && poolSize > 0 ? poolSize : 5;
> ```

5 - should be constant. 
___
*[`960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/49821#gh-1244172438:-NZ37mRF2WFn3JrLKSGv:b5dgloi) ([raw file](https://github.com/harness/harness-core/blob/1ca6120bf9a11c8be765c14621e5eca17f4bfe6a/960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java#L31)):*
<details><summary><i>Previously, markoharness (Marko Lukic) wrote…</i></summary><blockquote>

No. i will change the logging to be in debug mode (so that it can be use only when necessary). So that way we can leave the info we are logging as it is. What do you think?
</blockquote></details>

Do we really need this interceptor? I think when we turn on debug level on Azure SDK we will have all of this info related to API calls/requests in logging files, we can test and verify it.
https://learn.microsoft.com/en-us/azure/developer/java/sdk/logging-overview. We can set up AZURE\_LOG\_LEVEL env variable.

**Comment by markoharness** [comment]
*[`960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java` line 347 at r4](https://reviewable.io/reviews/harness/harness-core/49821#-NZ37RUQ6Zuu8pYylVVg:-NZ56PGN6WjJLKs7m4_c:b-fwj0th) ([raw file](https://github.com/harness/harness-core/blob/64f419a4bff7e0362b778b9bfd6828fcb953a2cb/960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java#L347)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Do we need a connection pool for each instance of the async client? Should we manage this connection pool, close and terminate connection pool threads when the API call is done?
</blockquote></details>

This is a copy of code from Azure SDK libs. 
And this connection pool is created twice (for two caches) as we are keeping it in memory.
(cachedAzureHttpClientForLogStreaming and cachedAzureHttpClient)

**Comment by markoharness** [comment]
*[`960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java` line 347 at r4](https://reviewable.io/reviews/harness/harness-core/49821#-NZ37caz5UhNujQk_ka1:-NZ56qhTC2G2PYKRvF9W:b-mo6ho4) ([raw file](https://github.com/harness/harness-core/blob/64f419a4bff7e0362b778b9bfd6828fcb953a2cb/960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java#L347)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

5 - should be constant. 
</blockquote></details>

Sure. I will change it.

**Comment by markoharness** [comment]
*[`960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/49821#gh-1244172438:-NZ56w_b7cFVxXYIwFj7:b1ylyem) ([raw file](https://github.com/harness/harness-core/blob/1ca6120bf9a11c8be765c14621e5eca17f4bfe6a/960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java#L31)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Do we really need this interceptor? I think when we turn on debug level on Azure SDK we will have all of this info related to API calls/requests in logging files, we can test and verify it.
https://learn.microsoft.com/en-us/azure/developer/java/sdk/logging-overview. We can set up AZURE\_LOG\_LEVEL env variable.
</blockquote></details>

Tried with env var AZURE_LOG_LEVEL=VERBOSE for delegate on QA and it is not logging anything.

**Review by vlica-harness — APPROVED** [review]

**Comment by markoharness** [comment]
trigger uts

**Review by ivan-83 — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49821)* status: 8 file(s) to review. 1. unresolved discussions. 

___
*[`960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java` line 347 at r4](https://reviewable.io/reviews/harness/harness-core/49821#-NZ37RUQ6Zuu8pYylVVg:-NZ7dbynAZcRlV26fdOw:bhvh8rs) ([raw file](https://github.com/harness/harness-core/blob/64f419a4bff7e0362b778b9bfd6828fcb953a2cb/960-api-services/src/main/java/io/harness/azure/utility/AzureUtils.java#L347)):*
<details><summary><i>Previously, markoharness (Marko Lukic) wrote…</i></summary><blockquote>

This is a copy of code from Azure SDK libs. 
And this connection pool is created twice (for two caches) as we are keeping it in memory.
(cachedAzureHttpClientForLogStreaming and cachedAzureHttpClient)

</blockquote></details>

Sure, this is a private static method and we should be careful here not to make it public. 
___
*[`960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/49821#gh-1244172438:-NZ7cfU9CoTB7D67s82Q:bthhrgx) ([raw file](https://github.com/harness/harness-core/blob/1ca6120bf9a11c8be765c14621e5eca17f4bfe6a/960-api-services/src/main/java/io/harness/azure/utility/LoggingInterceptor.java#L31)):*
<details><summary><i>Previously, markoharness (Marko Lukic) wrote…</i></summary><blockquote>

Tried with env var AZURE_LOG_LEVEL=VERBOSE for delegate on QA and it is not logging anything.
</blockquote></details>

Sure. It's better to push this logging info on DEBUG level as well as "Receive response ...". This could be useful for debugging purposes if we need to monitor response time.

**Comment by markoharness** [comment]
trigger review

**Pull request merged**